service/matching/tasklist/task_list_manager.go

```diff
@@ -268,49 +293,116 @@ func (c *taskListManagerImpl) handleErr(err error) error {
 	return err
 }
 
-func (c *taskListManagerImpl) loadTaskListPartitionConfig() {
-	if c.taskListID.IsRoot() {
-		return
-	}
+func (c *taskListManagerImpl) TaskListPartitionConfig() *types.TaskListPartitionConfig {
 	c.partitionConfigLock.RLock()
 	if c.partitionConfig != nil {
+		config := *c.partitionConfig
 		c.partitionConfigLock.RUnlock()
-		return
+		c.logger.Debug("get task list partition config from db", tag.Dynamic("root-partition", c.taskListID.GetRoot()), tag.Dynamic("config", config))
```

Comment:
the log key `config` may conflict with other log structs using the same key in search indices and will result in the entire log-line being dropped. I'd just make it more unique with a name like `tasklist-partition-config` to avoid this.